jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarModeProcessor.java

```diff
@@ -191,4 +202,49 @@ private static FileEntriesLayer addDirectoryContentsToLayer(
             });
     return builder.build();
   }
+
+  private static List<FileEntriesLayer> getDependenciesLayers(
+      Path jarPath, AbsoluteUnixPath pathOnContainer) throws IOException {
+    List<FileEntriesLayer> layers = new ArrayList<>();
+    String classPath = null;
+
+    // Get dependencies from Class-Path in the jar's manifest and add a layer each for non-snapshot
+    // and snapshot dependencies. If Class-Path is not present in the jar's manifest then skip
+    // adding the dependencies layers.
+    try (JarFile jarFile = new JarFile(jarPath.toFile())) {
+      classPath = jarFile.getManifest().getMainAttributes().getValue(Attributes.Name.CLASS_PATH);
+    }
+    if (classPath != null) {
+      Predicate<String> isSnapshot = name -> name.contains("SNAPSHOT");
```

Comment:
Now that we have refactored out this logic to this method, short-circuiting is possible, and I think it helps readability. And now I think it should be fine to expand the `try` block until the end. For example,

```suggestion
    try (JarFile jarFile = new JarFile(jarPath.toFile())) {
      String classPath = jarFile.getManifest().getMainAttributes().getValue(Attributes.Name.CLASS_PATH);
      if (classPath == null) {
        return new ArrayList<>();
      }

      List<FileEntriesLayer> layers = new ArrayList<>();
      ...
    }
```